.ci/xenial.jenkinsfile

```diff
@@ -28,14 +28,20 @@ def buildImage(String latestMaster, String latestBranch, String expectedImage, S
     sh "docker push ${latestBranch}"
 }
 
+
 def get_container_id() {
     // This is quite fragile and might depends on the Jenkins configuration and deployment, but it is the best
     //  I can figure out to get the ID of the running container to be able to mount the same volume when 
     //  using docker-in-docker.
     String nodeName = "${NODE_NAME}"
     String output = sh(script: 'docker ps --no-trunc', returnStdout: true)
-    String line = output.split('\n').each({ it.trim() }).findAll({ it.contains(nodeName) && it.contains('/opt/entrypoint.sh') })[0]
-    String nodeId = line.split(' ')[0]
+    String nodeId = null
+    output.split('\n').each({
+        String line = it.trim()
+        if (line.contains(nodeName) && line.contains('/opt/entrypoint.sh')) {
+            nodeId = line.split(' ')[0]
+        }
+    })
```

Comment:
`findAll` returns an iterator, which is not serializable. Depending on where the source is executed, this can lead to SerializationErrors in Jenkins.

Reply:
Good to know!